### PR TITLE
Allow specifying TTL when calling generate_cert.

### PR DESCRIPTION
**Pull request by myidpt**
Replace NotAfter with TTL. When NotBefore is not specified, generate_cert generates the cert relying on the current time. NotAfter will be the current time + TTL.

### CODE REVIEW

File: security/pkg/pki/ca/generate_cert.go [hunk 1]
```diff
@@ -46,8 +46,11 @@ type CertOptions struct {
 	// like kubernetes service account.
 	Host string
 
-	// The validity bounds of the issued certificate.
-	NotBefore, NotAfter time.Time
+	// The NotBefore field of the issued certificate.
+	NotBefore time.Time
+
+	// TTL of the certificate. NotAfter - NotBefore
+	TTL time.Duration
```
Comment (wattli) [inline security/pkg/pki/ca/generate_cert.go]: Should we give a default value of this?

File: security/pkg/pki/ca/generate_cert_test.go [hunk 2]
```diff
@@ -129,18 +129,16 @@ func TestLoadSignerCredsFromFiles(t *testing.T) {
 }
 
 func TestGenCert(t *testing.T) {
-	// set "notBefore" to be 5 minutes ago, this ensures the issued certifiate to
+	// set "notBefore" to be an hour ago, this ensures the issued certifiate to
```
Comment (wattli) [inline security/pkg/pki/ca/generate_cert_test.go]: Use 'one hour' instead?

Comment (myidpt) [reply security/pkg/pki/ca/generate_cert_test.go]: Done.

File: security/pkg/pki/ca/generate_cert.go (hunk 1 above)
Comment (myidpt) [reply security/pkg/pki/ca/generate_cert.go]: Yes, we do have the default values for the TTLs. The default values are given at the CA and node agent (as flags), this CertOptions is always initialized on CA and node agent.

**Pull request merged**